### DISCUSSION TITLE
關於 Client LocalStorage

### DISCUSSION THREAD

**Discussion by arsfy** [walinejs/waline#718] [Q&A] [answered]
請問 Client 有除了 LocalStorage 之外的其他方法設定使用者名稱/Email資訊嗎

**Comment by lizheming** [comment]
请问下需求是什么？

**Reply by arsfy** [reply]
> 请问下需求是什么？

我需要在 Android WebView 內使用 Waline，但是使用本地HTML檔案無法載入 LocalStorage
(我在Telegram Group(2021年6月) 詢問過如何使用外部程式碼傳入使用者資訊)

**Reply by lizheming** [reply]
如果可以在 HTML 档案中增加 Waline 的初始化脚本的话，应该是可以再加入 LocalStorage 写入的 JS 脚本？

**Reply by arsfy** [reply]
最後還是因為其他問題太多（奇怪的錯誤，如IP速率限制關閉時還是會觸發限制，客戶端有時候會不顯示頭相和使用者名稱，我想棄用WebView等....）決定自己造輪子

**Reply by arsfy** [reply]
速率限制一直有問題，原本以為可能是未校驗 X-Forwarded-For，打算關掉也會提示too fast

**Reply by lizheming** [reply]
你说的这些问题应该和 client 关系不大，可以单独提 issue 提供下环境我看看~不展示名称如果是发布评论后的话应该是之前的一个bug，你升级到最新版应该就好了

**Reply by arsfy** [reply]
不展示名稱這個問題我之前詢問過你，在upgrade到最新版本後還是可能會出現....

**Reply by arsfy** [reply]
我會在稍後提交issue

**Comment by Mister-Hope** [comment, marked as answer]
很抱歉回复较晚，还是补充一句，webview的确缺失了一些正常客户端有的全局变量，但你可以在调用 waline 前自行通过js 脚本 mock 这些全局变量
